airflow/api_connexion/parameters.py

```diff
@@ -86,3 +87,32 @@ def wrapped_function(*args, **kwargs):
         return cast(T, wrapped_function)
 
     return format_parameters_decorator
+
+
+def apply_sorting(model, query, order_by, to_replace=None):
+    """Apply sorting to query"""
+    if to_replace:
+        for key, value in to_replace.items():
+            if key == order_by:
+                order_by = value
+    if order_by.strip('-') not in (i.name for i in model.__table__.columns):
```

Comment:
It is not. looks safe because it allows you to sort on any attribute of an object, also on one that is hidden. e.g. password, account creation date. This way, we may be vulnerable to password leakage.  

Some attributes have a different name in the API and others in the API object, eg in REST APII, we use connection_id, and in the database conn_id. I think it's worth using one field name convention in the API.

Reply:
I wonder if we should have a definition of what models, and what fields on each model can be filtered on, either on the model classes themselves, or in a separate mapping maintained in `api_connexion`. (It should be a positive list so data won’t be leaked when a new model field is added.)

Reply:
I decided to maintain a list of allowed attributes on the endpoint since some models are flask appbuilder models. Let me know if this is good.

Another concern I have is that some endpoints were ordered by 'id' whereas 'id' is not in their openapi schema definition.  Removing 'id' from the list of attributes to order by in these endpoints will be a breaking change. For such endpoints, I retained a default order by 'id'. Let me know if this is OK @mik-laj  @uranusjr